### PR TITLE
[Feature:InstructorUI] Which individuals on team viewed grade

### LINKED ISSUE #3415: Feature request: which individuals on team have viewed grade
related to #3131 

Right now it just shows a green check mark, but it would be good to know which team member(s) have viewed their grade.  

Step 1:  We may need to modify the database to store more detailed information?    What's the right design?

Step 2:  Visually, how should we display this?  My suggestion...  red x = no one has seen it.  then a mix of green & grey checkmarks, if it's a team of 3 and 2 have viewed it will be 2 green checks and 1 grey check.  Then when you hover over the checks, the hover text lists the members who have viewed and the members who have not viewed.   Note:  If you hover over any single check it shows the same complete information for the team.  You don't have to hover over individual checkmarks.

Then on the status page, it lists the # of teams that have had at least one member view (current data), and separately the # of students who have viewed their team's grade.

### PR DESCRIPTION
**Pull request by jmcknight98**
Closes #3415
### Please check if the PR fulfills these requirements:

* [x] The PR title and message follows our [guidelines](https://submitty.org/developer/how_to_contribute)

### What is the current behavior?
- When any individual on a team views their grade, the instructor sees a green check for "Viewed Grade" column on the gradeable's details page.
- When any individual on a team views their grade, the "View Grade" button for students turns white even if they were not the one who viewed their grade.

### What is the new behavior?
- For team assignments, shows instructor a check mark for each student representing if they viewed their grade (Still shows a single red x if nobody has viewed grade as before). 
-  Hovering over the checks displays the time each student on the team viewed the assignment or if they have not. 
-  Information on the status page now displays the percentage of teams that have had at least one student view their grade and the percentage of students that have viewed their grade. 
- View Grade button for students is now green on team assignments until they view the grade even if a team member already has viewed it.
- Also added hover text for "Graded Questions" column.  Each circle has the graded component title.
checkmarkpic

### CODE REVIEW

File: site/app/libraries/database/DatabaseQueries.php [hunk 1]
```diff
@@ -1109,6 +1109,49 @@ public function getNumUsersWhoViewedGradeBySections($gradeable, $sections) {
...
+
+    /**
+     * @param $gradeable_id
+     * @param $team_id
+     * @return integer
+     */
+    public function getActiveVersionForTeam($gradeable_id,$team_id) {
+        $params = array($gradeable_id,$team_id);
+        $this->course_db->query("SELECT active_version FROM electronic_gradeable_version WHERE g_id = ? and team_id = ?",$params);
+        return $this->course_db->row()['active_version'];
```
Comment (sak6lab) [inline site/app/libraries/database/DatabaseQueries.php]: If a team has not made a submission there will be nothing returned. Include a check to make sure there are results returned.

File: site/public/css/server.css [hunk 2]
```diff
@@ -1370,6 +1370,31 @@ end of styles used in the admin gradeable page
     white-space: nowrap;
 }
 
+/* prevents line break in the grade viewed check marks */
+.td-viewed-grade {
+    white-space: nowrap;
+}
+
+.viewed-green-check {
```
Comment (sak6lab) [inline site/public/css/server.css]: Maybe some margin on the right and left but most left would look nice as the checks are hugging the left border.
hugging_checks

File: site/app/libraries/database/DatabaseQueries.php (hunk 1 above)
Comment (jmcknight98) [reply site/app/libraries/database/DatabaseQueries.php]: I added a check here as suggested in the latest commit and it appears to fix the issue.

File: site/public/css/server.css (hunk 2 above)
Comment (jmcknight98) [reply site/public/css/server.css]: Added some padding to the right, I messed around with it on the left as well and don't think it is necessary.  Does it look better now? I can update it more if there are any other suggestions.

File: migration/migrator/migrations/course/20190710182514_add_view_date_for_teams.py [hunk 3]
```diff
@@ -0,0 +1,5 @@
+"""Migration for a given Submitty course database."""
+
+def up(config, database, semester, course):
+    if not database.table_has_column('teams', 'last_viewed_time'):
+        database.execute('ALTER TABLE teams ADD COLUMN last_viewed_time timestamp with time zone')
```
Comment (bmcutler) [inline migration/migrator/migrations/course/20190710182514_add_view_date_for_teams.py]: Please add a down function with an empty body (just 'pass')
See other migrations for examples.

Comment (jmcknight98) [reply migration/migrator/migrations/course/20190710182514_add_view_date_for_teams.py]: Added in last commit.

Comment (jmcknight98) [reply migration/migrator/migrations/course/20190710182514_add_view_date_for_teams.py]: Added in last commit.

**Pull request merged**